android/runtime/v8/src/native/JSException.cpp

```diff
@@ -1,52 +1,82 @@
  */
+#include <cstring>
+#include <sstream>
+
 #include <jni.h>
 #include <v8.h>
 
 #include "JNIUtil.h"
 #include "TypeConverter.h"
+#include "V8Util.h"
 
 #include "JSException.h"
 
+#define MAX_STACK 10
+
 using namespace v8;
 
 namespace titanium {
 
 Local<Value> JSException::fromJavaException(v8::Isolate* isolate, jthrowable javaException)
 {
-	JNIEnv *env = JNIScope::getEnv();
+	JNIEnv* env = JNIScope::getEnv();
 	if (!env) {
 		return GetJNIEnvironmentError(isolate);
 	}
 
-	env->ExceptionDescribe();
-
 	bool deleteRef = false;
 	if (!javaException) {
 		javaException = env->ExceptionOccurred();
-		env->ExceptionClear();
 		deleteRef = true;
 	}
+	env->ExceptionClear();
 
-	//env->ExceptionDescribe();
-
-	jstring message = (jstring) env->CallObjectMethod(javaException, JNIUtil::throwableGetMessageMethod);
-	if (!message) {
+	jstring javaMessage = (jstring) env->CallObjectMethod(javaException, JNIUtil::throwableGetMessageMethod);
+	if (!javaMessage) {
 		return THROW(isolate, "Java Exception occurred");
 	}
 
-	Local<Value> jsMessage = TypeConverter::javaStringToJsString(isolate, env, message);
-	env->DeleteLocalRef(message);
+	Local<Context> context = isolate->GetCurrentContext();
+
+	// Grab the top-level error message
+	Local<Value> jsMessage = TypeConverter::javaStringToJsString(isolate, env, javaMessage);
+	env->DeleteLocalRef(javaMessage);
+	// Create a JS Error holding this message
+	// We use .As<String> here because we know that the return value of TypeConverter::javaStringToJsString
+	// must be a String. Only other variant is Null when the javaMessage is null, which we already checked for above.
+	// We use .As<Object> on Error because an Error is an Object.
+	Local<Object> error = Exception::Error(jsMessage.As<String>()).As<Object>();
 
+	// Now loop through the java stack and generate a JS String from the result and assign to Local<String> stack
+	std::stringstream stackStream;
+	jobjectArray frames = (jobjectArray) env->CallObjectMethod(javaException, JNIUtil::throwableGetStackTraceMethod);
+	jsize frames_length = env->GetArrayLength(frames);
+	for (int i = 0; i < (frames_length > MAX_STACK ? MAX_STACK : frames_length); i++) {
+		jobject frame = env->GetObjectArrayElement(frames, i);
+		jstring javaStack = (jstring) env->CallObjectMethod(frame, JNIUtil::stackTraceElementToStringMethod);
+
+		const char* stackPtr = env->GetStringUTFChars(javaStack, NULL);
+	 	stackStream << std::endl << "    " << stackPtr;
+
+		env->ReleaseStringUTFChars(javaStack, stackPtr);
+		env->DeleteLocalRef(javaStack);
+	}
 	if (deleteRef) {
 		env->DeleteLocalRef(javaException);
 	}
+	stackStream << std::endl;
+	Local<String> stack = String::NewFromUtf8(isolate, stackStream.str().c_str());
+
+	// Now explicitly assign our properly generated stacktrace
+	error->Set(context, STRING_NEW(isolate, "stack"), stack);
 
-	return isolate->ThrowException(jsMessage->ToString(isolate));
+	// throw it
+	return isolate->ThrowException(error);
```

Comment:
@sgtcoolguy The Java stack trace is now not shown in the log:
```
/app.js:9
       Ti.Geolocation.accuracy = null;
                               ^
   at (/app.js:9:33)
   at value(ti:/events.js:49:21)
   at value(ti:/events.js:101:19)

Uncaught Error: Unable to convert null
```

I appended it to the message as `ExceptionMessage` does not contain a stack element